src/webapp/forms.py

```diff
@@ -8,6 +8,94 @@
+        ("Severe", "Severe (most services down)"),
+        ("Intermittent Outage", "Intermittent Outage (may be up for some of the time)"),
+        ("No Significant Outage Expected", "No Significant Outage Expected (you shouldn't notice)")
+    ])
+    description = StringField("Description (the reason and/or impact of the outage)", [InputRequired()])
+
+    start_date = DateField("Start Date/Time (UTC)", [InputRequired()])
+    start_time = TimeField("&nbsp;", [InputRequired()])
+    end_date = DateField("End Date/Time (UTC)", [InputRequired()])
+    end_time = TimeField("&nbsp;", [InputRequired()])
+
+    facility = SelectField("Facility", choices=[], default="")
+    change_facility = SubmitField()
+    site = SelectField("Site", choices=[], default="")
+    change_site = SubmitField()
+    resource_group = SelectField("Resource Group", choices=[], default="")
+    change_resource_group = SubmitField()
+
+    generate = SubmitField()
+
+    yamloutput = TextAreaField(None, render_kw={"readonly": True,
+                                                "style": "font-family:monospace; font-size:small;",
+                                                "rows": "15"})
+
+    class Meta:
+        csrf = False  # CSRF not needed because no data gets modified
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.infos = ""
+
+    # https://stackoverflow.com/a/21815180
+    def validate(self):
+        self.infos = ""
+
+        if not super().validate():
+            return False
+        if self.start_date.data > self.end_date.data:
+            self.end_date.errors.append("End date/time must be after start date/time")
+            return False
+        elif self.start_date.data == self.end_date.data:
+            if self.start_time.data >= self.end_time.data:
+                self.end_time.errors.append("End date/time must be after start date/time")
+                return False
+
+        days_in_future = (self.get_start_datetime() - datetime.datetime.utcnow()).days
+        if days_in_future < 1 and self.scheduled.data == "SCHEDULED":
+            self.infos += "Note: Downtime registered less than one day in advance " \
+                             "is considered unscheduled by WLCG policy."
+        elif days_in_future >= 1 and self.scheduled.data == "UNSCHEDULED":
+            self.infos += "Note: Downtime registered at least one day in advance " \
+                             "is considered scheduled by WLCG policy."
+
+        return True
+
+    def get_start_datetime(self):
+        return datetime.datetime.combine(self.start_date.data, self.start_time.data)
+
+    def get_end_datetime(self):
+        return datetime.datetime.combine(self.end_date.data, self.end_time.data)
+
+    def get_yaml(self, resources, services) -> str:
+        yaml = ""
+        for resource, service in zip(resources, services):
+            yaml += models.get_downtime_yaml(
```

Comment:
The downtime entry for each _resource_ must have a unique ID -- I don't see any code for dealing with that.

I suggest:
- Modify `models.get_downtime_yaml()` to take an ID as an optional parameter
- Call `models._dtid()` before the loop to get the initial ID
- Pass the ID to `get_downtime_yaml()`
- Increment the ID with each iteration of the loop